azurerm/resource_arm_role_assignment.go

```diff
@@ -64,6 +64,13 @@ func resourceArmRoleAssignment() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
+
+			"skip_service_principal_aad_check": {
```

Comment:
thinking about this, I'm fine with this approach, but is there any way that we could look this information up so that users don't need to specify it?

Reply:
given this is all a hack around broken AAD replication, and that there might be other reasons to set the type property (or at the least know it) i'm more inclined to just expose the SDK/API type field and users can set it as desired.

Reply:
Per the service team:
* The only valid values that are publicly exposed by our APIs are: **User**, **Group**, and **ServicePrincipal**.

* This property is more relevant on the read path than write. **For writes, the only relevant scenario is for ServicePrincipal**. I see no value add in specifying `PrincipalType` as User or Group. The service anyways checks against AAD and fails the call if the principal is not found or the principal type in AAD doesn’t match what the request specified. Only for `ServicePrincipal`, if AAD returns principal not found, then role assignment create still proceeds in the new api-version.

* **Don’t set this parameter unless explicitly dealing with newly created service principal.**

* As stated above this attribute makes more sense in the read scenario, the API will return User, Group, ServicePrincipal based on the type of the principal in AAD.